src/java.base/share/classes/java/lang/StableValue.java

```diff
@@ -403,7 +400,20 @@
  * {@linkplain StableValue#orElseSet(Supplier) orElseSet()} they too are
  * thread safe and guarantee at-most-once-per-input invocation.
  *
- * @implSpec Implementing classes of {@linkplain StableValue} are free to synchronize on
+ * <h2 id="performance">Performance</h2>
+ * A stable value that is <em>set</em> is treated as a constant by the JVM, enabling the
+ * same performance optimizations that are available for {@code final} fields.
+ * As such, stable values can be used to replace {@code final} fields in cases where
+ * <em>at-most-once</em> update semantics is crucial, but where the eager initialization
+ * semantics associated with {@code final} fields is too restrictive.
+ * <p>
+ * In JDK 24, {@code final} instance fields in records and hidden classes (such as classes
```

Comment:
I suggest to drop this -- we normally don't reference to JDK versions in the javadoc.

Perhaps this para should say: 

```
The _content_ of a set stable value is treated as a constant by the JVM, provided that the reference to the stable value is also constant (e.g. in cases where the stable value itself is stored in a `static final` field). This means that, at least in some cases, access to the content of a stable value enjoys the same constant-folding optimizations that are available when accessing `static final` fields.
```